### DISCUSSION TITLE
Stylistic Sets up to ss99

### DISCUSSION THREAD

**Discussion by kenmcd** [psb1558/Junicode-font#181] [General]
Been watching your discussion over at https://github.com/psb1558/Junicode-font/issues/180 and how you are out of character variants space, and remembered how you are also running up to "limits" on stylistic sets - and thought about this again.

The Beloved Script and Sans fonts from Laura Worthington have stylistic sets all the way up to ss74.

Beloved Script has: ss01 ss02 ss03 ss04 ss05 ss06 ss07 ss08 ss09 ss10 ss11 ss12 ss13 ss14 ss15 ss16 ss17 ss18 ss19 ss20 ss21 ss22 ss23 ss24 ss25 ss26 ss27 ss28 ss29 ss30 ss31 ss32 ss33 ss34 ss35 ss36 ss37 ss38 ss39 ss40 ss41 ss42 ss43 ss44 ss45 ss46 ss47 ss48 ss49 ss50 ss51 ss52 ss53 ss54 ss55 ss56 ss57 ss58 ss59 ss60 ss61 ss62 ss63 ss64 ss65 ss66 ss67 ss68 ss69 ss70 ss71 ss72 ss73 ss74

Beloved Sans has: ss01 ss02 ss03 ss04 ss05 ss06 ss07 ss08 ss09 ss10 ss11 ss12 ss13 ss14 ss15 ss16 ss17 ss18 ss19 ss20 ss21 ss22 ss23 ss24 ss25 ss26 ss27 ss28 ss29 ss30 ss31 ss32 ss33 ss34 ss35 ss36 ss37 ss38

These fonts have been around since 2015.

She creates the very complete documentation in InDesign.
So I assume InDesign has worked fine since 2015.
Also working fine: Affinity Publisher, LibreOffice, Inkscape
Browsers tested: Firefox, and Vivaldi (chromium) - both work fine

I would guess it would work fine in any application which uses Harfbuzz.
You would need to test LaTeX.

So you may want to consider adding additional stylistic sets all the way up to ss99.
Which should help in adding new features (like Irish variants, etc.).

I can send you the docs and the fonts if you would like to take a look.
(or just links to the freely available doc downloads)

Example:
![UGuide-BelovedScript_1](https://github.com/psb1558/Junicode-font/assets/844920/25032d95-c5e6-4c6d-85f1-d671294a19e7)

**Comment by psb1558** [comment]
I'll do some testing. From my point of view, it needs to work with all major layout engines to be useful.

Extending the number of `ssNN` features up to 99 has been suggested to Microsoft (for better or worse the custodian of the OpenType spec), but it has been rejected, if I remember correctly because no more than 20 stylistic sets should be needed when they're used correctly.

This, and the general rigidity of the OT feature set (why shouldn't it allow custom features?), are a great frustration to some in the font development world.

Here's another thought, which would violate the prohibition against custom features but maintain the distinction between `cvNN` and `ssNN` features: simply decree that the `NN` in `cvNN` can be two hex digits, e.g. `cv1A`. I'll test this too.

I doubt that custom `ssNN` features can be inserted in InDesign via the UI. There are scripts that will probably do it.

**Reply by kenmcd** [reply]
> 
> I doubt that custom `ssNN` features can be inserted in InDesign via the UI.

No, it works in InDesign just like any other stylistic sets.

The only application I tested that had a limit to 20 was Word (of course). But who in their right mind would try to use Junicode advanced features in Word? An exercise in futility.
